tests/integration/test_charm.py

```diff
@@ -118,15 +128,60 @@ async def test_settings_are_correct(ops_test: OpsTest, series: str):
     assert settings["maximum_lag_on_failover"] == 1048576
 
 
-def build_application_name(series: str) -> str:
-    """Return a composite application name combining application name and series."""
-    return f"{APP_NAME}-{series}"
+@pytest.mark.parametrize("series", SERIES)
+async def test_scale_down_and_up(ops_test: OpsTest, series: str):
+    """Test data is replicated to new units after a scale up."""
+    # Set a composite application name in order to test in more than one series at the same time.
+    application_name = build_application_name(series)
+
+    # Ensure the initial number of units in the application.
+    initial_scale = len(UNIT_IDS)
+    await scale_application(ops_test, application_name, initial_scale)
+
+    # Scale down the application.
+    await scale_application(ops_test, application_name, initial_scale - 1)
+
+    # Ensure the member was correctly removed from the cluster
+    # (by comparing the cluster members and the current units).
+    await check_cluster_members(ops_test, application_name)
+
+    # Scale up the application (2 more units than the current scale).
+    await scale_application(ops_test, application_name, initial_scale + 1)
+
+    # Assert the correct members are part of the cluster.
+    await check_cluster_members(ops_test, application_name)
 
 
-def convert_records_to_dict(records: List[tuple]) -> dict:
-    """Converts pyscopg2 records list to a dict."""
-    dict = {}
-    for record in records:
-        # Add record tuple data to dict.
-        dict[record[0]] = record[1]
-    return dict
+@pytest.mark.parametrize("series", SERIES)
+async def test_persist_data_through_leader_deletion(ops_test: OpsTest, series: str):
```

Comment:
nit change this to `test_persist_data_through_primary_deletion`

Reply:
Fixed on https://github.com/canonical/postgresql-operator/pull/14/commits/3cadd303971eda2ab85ffa8e57840b176d66e9fb.